frontend/src/components/SeedlotRegistrationSteps/OrchardStep/index.tsx

```diff
@@ -205,92 +194,71 @@ const OrchardStep = ({ setStep }: OrchardStepProps) => {
     setResponse([name], checked);
   };
 
-  const percentageHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const { name, value } = event.target;
-    setResponse([name], [value]);
-  };
-
   const validateBreedingPercentage = () => {
-    const intNumber = +responseBody.breedingPercentage;
+    const intNumber = +state.breedingPercentage;
     if (intNumber < 0
-        || intNumber > 99
-        || !responseBody.breedingPercentage) {
+      || intNumber > 99
+      || !state.breedingPercentage) {
```

Comment:
This code looks good, but how about using `parseInt` and `isNan` here? Makes the code more elegant :)

Reply:
@mgaseta you're the author for this piece, what do you think?
Personally I like to use `Number()` instead of `parseInt()` 😝, this [article](https://thisthat.dev/number-constructor-vs-parse-int/#:~:text=Number()%20converts%20the%20type%20whereas%20parseInt%20parses%20the%20value%20of%20input.&text=As%20you%20see%2C%20parseInt%20will,parseInt%20accepts%20two%20parameters.) suggests that we use the `+` operator though.

Also, is 100 not allowed?

Reply:
I used the `+` operator already a few lines up, the `|| !state.breedingPercentage` is just to check if the field is empty, since the number input component will already block any "non-number" digits, so not sure if `isNaN` is the best option here since it can be only empty `''` or a number, but also if we need to remove these validations, maybe this can be removed/commented for now?

Not sure why I used 99 there, it should be 100... Sorry about that.

Reply:
Interesting! Thanks for sharing the link @craigyu No need to update. JavaScript indeed has a lot of options for that 😄

Reply:
the upper range limit has been updated with this [commit](https://github.com/bcgov/nr-spar-webapp/pull/82/commits/77d0c4a37f37d1560bf263e546f6934c47346e47)